execution/executor/src/components/chunk_output.rs

```diff
@@ -86,3 +93,96 @@ impl ChunkOutput {
         }
     }
 }
+
+pub fn update_counters_for_processed_chunk(
+    transactions: &[Transaction],
+    transaction_outputs: &[TransactionOutput],
+    process_type: &str,
+) {
+    let detailed_counters = AptosVM::get_processed_transactions_detailed_counters();
+
+    if transactions.len() != transaction_outputs.len() {
+        warn!(
+            "Chunk lenthgs don't match: txns: {} and outputs: {}",
+            transactions.len(),
+            transaction_outputs.len()
+        );
+    }
+
+    for (txn, output) in transactions.iter().zip(transaction_outputs.iter()) {
+        let status = match output.status() {
+            TransactionStatus::Keep(_) => "success",
+            TransactionStatus::Discard(_) => "failed",
+            TransactionStatus::Retry => "retry",
+        };
+
+        let kind = match txn {
+            Transaction::UserTransaction(_) => "user_transaction",
+            Transaction::GenesisTransaction(_) => "genesis",
+            Transaction::BlockMetadata(_) => "block_metadata",
+            Transaction::StateCheckpoint(_) => "state_checkpoint",
+        };
+
+        metrics::APTOS_PROCESSED_TXNS_COUNT
+            .with_label_values(&[process_type, kind, status])
+            .inc();
+
+        if let Transaction::UserTransaction(user_txn) = txn {
+            match user_txn.payload() {
+                aptos_types::transaction::TransactionPayload::Script(_script) => {
+                    metrics::APTOS_PROCESSED_USER_TRANSACTIONS_PAYLOAD_TYPE
+                        .with_label_values(&[process_type, "script"])
+                        .inc();
+                }
+                aptos_types::transaction::TransactionPayload::ModuleBundle(_module) => {
+                    metrics::APTOS_PROCESSED_USER_TRANSACTIONS_PAYLOAD_TYPE
+                        .with_label_values(&[process_type, "module"])
+                        .inc();
+                }
+                aptos_types::transaction::TransactionPayload::EntryFunction(function) => {
+                    metrics::APTOS_PROCESSED_USER_TRANSACTIONS_PAYLOAD_TYPE
+                        .with_label_values(&[process_type, "function"])
+                        .inc();
+
+                    let is_core = function.module().address() == &CORE_CODE_ADDRESS;
+                    metrics::APTOS_PROCESSED_USER_TRANSACTIONS_ENTRY_FUNCTION_MODULE
+                        .with_label_values(&[
+                            process_type,
+                            if is_core { "core" } else { "user" },
+                            if is_core || detailed_counters {
+                                function.module().name().as_str()
```

Comment:
How do we think about maintaining some sort of whitelisted modules/functions via configs and for them we can emit the details counter? That will give us the flexibility for us to only get counters for important functions we care about.

Reply:
@sitalkedia let's do that in a separate PR, to leave the complexity out of this one?